File: Kotlin/jumpGame.kt
class Solution {
    fun canJump(nums: IntArray): Boolean {
        if(nums.size < 2){return true}
        var energy = nums[0]
        for (i in (0..nums.size - 2)){
            if (nums[i] > energy){
                energy = nums[i]
            }
            if (energy == 0){
                return false
            }
            energy--
        }
        return true
    }
}

File: Python/Array/String/jumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) < 2:return True
        nums.pop()
        energy = nums[0]
        for num in nums:
            if num > energy:
                energy = num
            if energy == 0:
                return False
            energy -= 1
        return True